flink-staging/flink-streaming/flink-streaming-connectors/src/main/java/org/apache/flink/streaming/connectors/kafka/api/persistent/PersistentKafkaSource.java

```diff
@@ -236,13 +246,22 @@ public void restoreState(long[] state) {
 	@Override
 	public void commitCheckpoint(long checkpointId) {
 		LOG.info("Commit checkpoint {}", checkpointId);
-		long[] checkpointOffsets = pendingCheckpoints.remove(checkpointId);
-		if(checkpointOffsets == null) {
+		final int posInMap = pendingCheckpoints.indexOf(checkpointId);
+		if(posInMap == -1) {
 			LOG.warn("Unable to find pending checkpoint for id {}", checkpointId);
 			return;
 		}
+
+		long[] checkpointOffsets = (long[]) pendingCheckpoints.remove(posInMap);
 		LOG.info("Got corresponding offsets {}", Arrays.toString(checkpointOffsets));
```

Comment:
I think this log message can be improved. To something like "Committing offsets {} to Kafka."